cpp/include/cudf/detail/utilities/device_operators.cuh

```diff
@@ -27,6 +27,10 @@
 #include <cudf/types.hpp>
 #include <cudf/utilities/error.hpp>
 #include <cudf/utilities/traits.hpp>
+
+// will fail to compile if grouped with the includes above
+#include <cudf/fixed_point/fixed_point.hpp>
```

Comment:
What's the error here? Is it to prevent `_LIBCUDACXX_USE_CXX17_TYPE_TRAITS` or `simt` being included before the above headers, or is it because `fixed_point.hpp` is missing some headers that need to be included before it?

Reply:
Are the simt headers being included by fixed_point blocking the inclusion of the regular ones?

Reply:
I will follow up and look into what the exact reason is - will either set up a follow up issue to address or provide a better solution in this PR if it is straight-forward.

Reply:
It seems that if the `fixed_point` header is included before the `type_traits` header then there is conflict in one of the `thrust` headers? 
```
/home/cph/cudf/cpp/build/_deps/thrust-src/thrust/type_traits/is_contiguous_iterator.h(105): error: namespace "simt::std::v1" has no member "__wrap_iter"
```

Reply:
Correction, it is actually the following two includes that cause a conflict:
```
#include <cudf/fixed_point/fixed_point.hpp>
#include <cudf/utilities/traits.hpp>
```
Including `fixed_point.hpp` in `<cudf/utilities/traits.hpp>` causes the following error:
```
In file included from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/iterator/detail/tagged_iterator.h:23:0,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/detail/temporary_array.h:38,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/system/detail/generic/scan_by_key.inl:25,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/system/detail/generic/scan_by_key.h:143,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/detail/scan.inl:27,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/scan.h:1563,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/system/detail/generic/reduce_by_key.inl:35,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/system/detail/generic/reduce_by_key.h:88,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/detail/reduce.inl:26,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/reduce.h:784,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/system/detail/generic/find.inl:19,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/system/detail/generic/find.h:62,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/detail/find.inl:25,
                 from /home/cph/cudf/cpp/build/_deps/thrust-src/thrust/find.h:384,
                 from /home/cph/cudf/cpp/include/cudf/strings/./string_view.inl:17,
                 from /home/cph/cudf/cpp/include/cudf/strings/string_view.cuh:352,
                 from /home/cph/cudf/cpp/include/cudf/utilities/traits.hpp:23,
                 from /home/cph/cudf/cpp/include/cudf/scalar/scalar.hpp:19,
                 from /home/cph/cudf/cpp/src/jit/type.cpp:18:
/home/cph/cudf/cpp/build/_deps/thrust-src/thrust/type_traits/is_contiguous_iterator.h:105:10: error: ‘__wrap_iter’ is not a member of ‘simt::std::v1’
   _VSTD::__wrap_iter<Iterator>
```
Looking into this now.

Reply:
Have setup https://github.com/rapidsai/cudf/issues/5746 to solve this at a later date.